src/providers/completer/citation.ts

```diff
@@ -120,11 +120,10 @@ export class Citation implements IProvider {
         })
     }
 
-    getEntryDict(): {[key: string]: Suggestion} {
+    getEntry(key: string): Suggestion | undefined {
         const suggestions = this.updateAll()
-        const entries: {[key: string]: Suggestion} = {}
-        suggestions.forEach(entry => entries[entry.key] = entry)
-        return entries
+        const entry = suggestions.find((elm) => elm.key === key)
```

Comment:
The function `find` returns the first element matching the condition whereas in the previous implementation we actually returned the last one if there were multiple matches. I am not if it can happen or even it matters.

Reply:
I think the change is acceptable.